lib/kubeclient/common.rb

```diff
@@ -286,6 +308,45 @@ def get_entities(entity_type, klass, resource_name, options = {})
       Kubeclient::Common::EntityList.new(entity_type, resource_version, collection)
     end
 
+    def get_entities_async(klass, resource_name, options = {})
+      params = {}
+      SEARCH_ARGUMENTS.each { |k, v| params[k] = options[v] if options[v] }
+
+      ns_prefix = build_namespace_prefix(options[:namespace])
+
+      resource_version = nil
+      entity_streamer = Json::Streamer::JsonStreamer.new
+      resource_version_streamer = Json::Streamer::JsonStreamer.new
+
+      resource_version_streamer.get(key: 'resourceVersion') do |value|
+        resource_version = value unless resource_version
+      end
+
+      # result['items'] might be nil due to https://github.com/kubernetes/kubernetes/issues/13096
+      entity_streamer.get(nesting_level: 2, yield_values: false) do |object|
+        entity = new_entity(object, klass)
+        yield(entity, resource_version)
+      end
+
+      block = proc do |response|
+        if response.instance_of?(Net::HTTPNotFound)
+          raise ResourceNotFoundError.new(404,
+                                          'Not Found',
+                                          response)
+        end
+        response.read_body do |chunk|
+          entity_streamer.parser << chunk
```

Comment:
Shouldn't you also feed it `resource_version_streamer`?  AFAICT it's never fed, so `resource_version` will remain nil.  Are you testing yielded resource_version and do tests pass?

P.S. 2 streamers means everything gets parsed twice?  Or will the `resource_version_streamer` know its work is done as soon as it saw `resource_version`?

Reply:
updated

Reply:
yes; if json-streamer is updated it would be a nice improvement. however at this point it doesnt seem like much overhead, as the 2 other streamers will be 'deactivated' once they match on a single key (and for the time being, those keys are present at the  start of the json object we're parsing).